src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -14,14 +14,14 @@ namespace Microsoft.AspNetCore.Razor.Runtime.TagHelpers
     /// </summary>
     public class TagHelperExecutionContext
     {
-        private readonly string _tagName;
-        private readonly string _uniqueId;
-        private readonly TagMode _tagMode;
         private readonly List<ITagHelper> _tagHelpers;
-        private readonly Func<Task> _executeChildContentAsync;
         private readonly Action<HtmlEncoder> _startTagHelperWritingScope;
         private readonly Func<TagHelperContent> _endTagHelperWritingScope;
```

Comment:
Too much understanding of how MVC uses the scope manager here. The start and end bits (`Action<HtmlEncoder>` and `Func<TagHelperContent>`) are passed into the `Begin()` method and should not be treated as invariant here. Update `TagHelperScopeManager` and code gen to pass these delegates into the `TagHelperScopeManager` constructor. Then passing them into this class' constructor obviously be fine.